### PR TITLE
Adding optional confirm_req answers

### PR DESCRIPTION
**Pull request by SergiySW**
If tthis is request for forked block/account representative knows about

### CODE REVIEW

**Comment by PlasmaPower** [comment]
`node.ledger.successor` should handle a lot of this logic for you.

**Comment by SergiySW** [comment]
Yeah, testing with this

**Comment by PlasmaPower** [comment]
I think you misunderstood what I meant. `node.ledger.successor` will automatically detect if the root is an account or a block, and return said block. You will no longer have to split up the conditions.

**Comment by SergiySW** [comment]
Hm, ok

**Comment by SergiySW** [comment]
This should be valid now

**Comment by PlasmaPower** [comment]
LGTM

**Comment by PlasmaPower** [comment]
Actually, I think you'll have to revert the last two commits :(. Even if a block's root exists, the block itself (or another successor) may not exist, which causes a crash.

**Comment by SergiySW** [comment]
Hm, there is check for nullptr, it what conditions it can cause crash?

**Comment by PlasmaPower** [comment]
`node.ledger.successor` asserts that it's not nullptr, so it'll crash in debug mode. Maybe you should add a function that's like `node.ledger.successor` but that returns an optional?

**Comment by PlasmaPower** [comment]
LGTM again.

File: rai/ledger.cpp [hunk 1]
```diff
@@ -865,9 +863,7 @@ std::unique_ptr<rai::block> rai::ledger::successor (MDB_txn * transaction_a, rai
 	{
 		successor = store.block_successor (transaction_a, block_a);
```
Comment (clemahieu) [inline rai/ledger.cpp]: Successor might not be initialized.

**Pull request merged**